pyw3d/objects.py

```diff
@@ -252,8 +252,94 @@ def fromXML(content_root):
             return W3DLight.fromXML(content_root)
         if content_root.find("ParticleSystem") is not None:
             return W3DPSys.fromXML(content_root)
+        if content_root.find("Shape") is not None:
+            return W3DShape.fromXML(content_root)
         raise BadW3DXML("No known child node found in Content node")
 
+class W3DShape(W3DContent):
+    """ Create a shape object in virtual space
+
+    :param str shape_type: type of shape to be created
+    :param int radius: radius of the shape to be added
+    :param tuple color: Three floats representing RGB color of object 
+    :param int depth: height of shape, if shape is cone/cylinder
+    """
+    ui_order = ["shape_type","radius", "rotation", "depth"]
+    argument_validators = {
+        "shape_type": OptionValidator("Sphere", "Cube", "Cone", "Cylinder", "Monkey"),
+        "radius": IsNumeric(),
+        "rotation":ListValidator(IsInteger(min_value=0, max_value=255), required_length=3),
+        "depth": IsNumeric()
+    }
+    default_arguments = {
+        "radius": 1,
+        "rotation": (0,0,0),
+        "depth":2
+    }
+    ui_order = ["shape_type","radius", "rotation", "depth"]
+
+    def toXML(self, object_root):
+
+        content_root = ET.SubElement(object_root, "Content")
+        attrib = {
+            "shape_type": self["shape_type"],
+            "radius": self["radius"],
+            "rotation": self["rotation"],
+            "depth": self["depth"]
+        }
```

Comment:
I heartily approve of this XML scheme. Clean, concise, and intuitive, unlike a lot of what we already have!